wazo_call_logd/participant.py

```diff
@@ -1,15 +1,71 @@
 # Copyright 2021 The Wazo Authors  (see the AUTHORS file)
 # SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import annotations
 
 import logging
-
+import requests.exceptions
 from xivo.asterisk.protocol_interface import protocol_interface_from_channel
 from xivo.asterisk.protocol_interface import InvalidChannelError
+from wazo_confd_client import Client as ConfdClient
+from typing import NamedTuple
 
 logger = logging.getLogger(__name__)
 
 
-def find_participant(confd, channame):
+class ParticipantInfo(NamedTuple):
+    uuid: str
+    tenant_uuid: str
+    line_id: int | None
+    tags: list[str]
+    main_extension: str | None
+
+
+def get_tags(field: str | None) -> list[str]:
+    return [tag.strip() for tag in field.split(',')] if field else []
+
+
+def find_participant_by_uuid(confd, user_uuid: str) -> ParticipantInfo | None:
+    try:
+        user = confd.users.get(user_uuid)
+    except requests.exceptions.HTTPError:
+        logger.exception("Error retrieving user(user_uuid=%s) from confd", user_uuid)
+        return None
+
+    if not user:
+        return None
+
+    tags = get_tags(user['userfield'])
+    logger.debug(
+        'Found participant with user uuid %s, tenant uuid %s',
+        user['uuid'],
+        user['tenant_uuid'],
+    )
+    # NOTE(charles): this replicates the behavior of CallLogsGenerator._update_call_participants_with_their_tags
+    # of defaulting to first line of user, though this might be erroneous
+    line = user['lines'] and user['lines'][0]
+    logger.debug("user(user_uuid=%s) has first line: %s", user_uuid, line)
+    main_extension = (
+        line
+        and ('extensions' in line or None)
+        and (line['extensions'] or None)
+        and line['extensions'][0]
+    )
```

Comment:
I believe I added the check for the `extensions` key because a test failed on this attribute. I'll see about changing the tests mock instead.

The confd client is untyped and has no schema validation, so at the code level little is guaranteed. Existing tests did not use mock data that matched the API contract. Exporting schema definitions in client libraries would enable more explicit interfaces and guide(perhaps help autogenerate) better mock data for tests.

Reply:
And indeed, integration tests are the reasons for this check of `extensions` key.